Basic/ClientDriven/Assets/Scripts/HostJoinUI.cs

```diff
@@ -0,0 +1,115 @@
+using System;
+using System.Text.RegularExpressions;
+using Unity.Netcode;
+using Unity.Netcode.Transports.UTP;
+using UnityEngine;
+using UnityEngine.UIElements;
+
+public class HostJoinUI : MonoBehaviour
+{
+    [SerializeField]
+    UIDocument m_MainMenuUIDocument;
+
+    [SerializeField]
+    UIDocument m_InGameUIDocument;
+
+    VisualElement m_MainMenuRootVisualElement;
+    
+    VisualElement m_InGameRootVisualElement;
+    
+    Button m_HostButton;
+    
+    Button m_ServerButton;
+    
+    Button m_ClientButton;
+
+    TextField m_IPAddressTextField;
+    
+    TextField m_PortTextField;
+    
+    void Awake()
+    {
+        m_MainMenuRootVisualElement = m_MainMenuUIDocument.rootVisualElement;
+        m_InGameRootVisualElement = m_InGameUIDocument.rootVisualElement;
+        
+        m_HostButton = m_MainMenuRootVisualElement.Query<Button>("HostButton");
+        m_ClientButton = m_MainMenuRootVisualElement.Query<Button>("ClientButton");
+        m_ServerButton = m_MainMenuRootVisualElement.Query<Button>("ServerButton");
+        m_IPAddressTextField = m_MainMenuRootVisualElement.Query<TextField>("IPAddressField");
+        m_PortTextField = m_MainMenuRootVisualElement.Query<TextField>("PortField");
+        
+        m_HostButton.clickable.clickedWithEventInfo += StartHost;
+        m_ServerButton.clickable.clickedWithEventInfo += StartServer;
+        m_ClientButton.clickable.clickedWithEventInfo += StartClient;
+    }
+
+    void Start()
+    {
+        ToggleMainMenuUI(true);
+        ToggleInGameUI(false);
+    }
+
+    public void StartHost(EventBase obj)
+    {
+        SetUtpConnectionData();
+        var result= NetworkManager.Singleton.StartHost();
```

Comment:
```suggestion
        var result = NetworkManager.Singleton.StartHost();
```
Same thing in the other methods below